Src/HALALMock/Services/Communication/FDCAN/FDCAN.cpp

```diff
@@ -89,43 +92,52 @@ bool FDCAN::transmit(uint8_t id, uint32_t message_id, const char* data, FDCAN::D
 		return false;
 	}
 
-	instance->tx_header.Identifier = message_id;
+    size_t buffer_len = sizeof(message_id)+ sizeof(dlc) + dlc;
+    char* temp_data = new char[buffer_len];
 
-	if (dlc != FDCAN::DLC::DEFAULT) {
-		instance->tx_header.DataLength = dlc;
-	}
+    temp_data[0] = (message_id >> 24);
+    temp_data[1] = (message_id >> 16);
+    temp_data[2] = (message_id >> 8);
+    temp_data[3] = message_id;
 
-	HAL_StatusTypeDef error = HAL_FDCAN_AddMessageToTxFifoQ(instance->hfdcan, &instance->tx_header, (uint8_t*)data);
+	temp_data[4] = ((uint32_t)dlc >> 24);
+	temp_data[5] = ((uint32_t)dlc >> 16);
+	temp_data[6] = ((uint32_t)dlc >> 8);
+	temp_data[7] = (uint32_t)dlc;
 
-	if (error != HAL_OK) {
+	memcpy((temp_data + sizeof(message_id)+ sizeof(dlc)), data, dlc);
+		
+	if(send(instance->socket,temp_data, strlen(data))<0){
```

Comment:
Might seem weird, but a single call to send might not send all data, you should place this in a loop and check how many bytes have been sent, like so:
```c++
ssize_t total_bytes_sent{0};
while (total_bytes_sent < strlen(temp_data)) {
    ssize_t bytes_sent = send(instance->socket, temp_data + total_bytes_sent, strlen(temp_data) - total_bytes_sent);
    // if (bytes_sent < 0 ...
    total_bytes_sent += bytes_sent
}
```
Seems weird if you don't know how the protocols behind it work but could happen and the documentation on send isn't very explicit about this